bots/csr/src/main/java/org/openjdk/skara/bots/csr/PullRequestWorkItem.java

```diff
@@ -144,81 +146,103 @@ public Collection<WorkItem> run(Path scratchPath) {
-            // If the PR body doesn't have the CSR issue or doesn't have the CSR progress,
-            // this bot need to add the csr update marker so that the PR bot can update the message of the PR body.
-            log.info("The PR body doesn't have the CSR issue or progress, adding the csr update marker for " + describe(pr));
-            addUpdateMarker(pr);
-        }
+        for (var issue : issues) {
+            var jbsIssueOpt = project.issue(issue.shortId());
+            if (jbsIssueOpt.isEmpty()) {
+                // One issue could not be found in JBS, so the csr label cannot be removed until the problem is solved
+                allCSRApproved = false;
+                var issueId = issue.project().isEmpty() ? (project.name() + "-" + issue.id()) : issue.id();
+                log.info("No issue found in JBS related with this issue " + issueId + "for " + describe(pr));
+                // allCSRApproved is false now, so we could break now
+                break;
+            }
 
-        var resolution = csr.properties().get("resolution");
-        if (resolution == null || resolution.isNull()) {
-            if (!pr.labelNames().contains(CSR_LABEL)) {
-                log.info("CSR issue resolution is null for " + describe(pr) + ", adding the CSR label");
-                pr.addLabel(CSR_LABEL);
-            } else {
-                log.info("CSR issue resolution is null for " + describe(pr) + ", not removing the CSR label");
+            var csrOptional = Backports.findCsr(jbsIssueOpt.get(), versionOpt.get());
+            if (csrOptional.isEmpty()) {
+                log.info("No CSR found for issue " + jbsIssueOpt.get().id() + " for " + describe(pr));
+                continue;
             }
-            logLatency();
-            return List.of();
-        }
-        var name = resolution.get("name");
-        if (name == null || name.isNull()) {
-            if (!pr.labelNames().contains(CSR_LABEL)) {
-                log.info("CSR issue resolution name is null for " + describe(pr) + ", adding the CSR label");
-                pr.addLabel(CSR_LABEL);
-            } else {
-                log.info("CSR issue resolution name is null for " + describe(pr) + ", not removing the CSR label");
+            var csr = csrOptional.get();
+
+            log.info("Found CSR " + csr.id() + " for issue " + jbsIssueOpt.get().id() + " for " + describe(pr));
+            if (!hasCsrIssueAndProgress(pr, csr) && !isWithdrawnCSR(csr)) {
+                // If the PR body doesn't have the CSR issue or doesn't have the CSR progress,
+                // this bot need to add the csr update marker so that the PR bot can update the message of the PR body.
+                log.info("The PR body doesn't have the CSR issue or progress, adding the csr update marker for this csr issue"
+                        + csr.id() + " for " + describe(pr));
+                addUpdateMarker(pr);
```

Comment:
`addUpdateMarker` isn't suitable to be called multiple times. Each call will modify the body again.

Reply:
Right, I will change it, thx!